rtCommon/bidsArchive.py

```diff
@@ -0,0 +1,825 @@
+                                     archiveImg.affine,
+                                     header=archiveImg.header)
+            newImg.update_header()
+            self._addImage(newImg, imgPath)
+
+        # 2.2) Image doesn't exist in archive, but rest of the path is valid for
+        # the archive; create new Nifti file within the archive
+        elif self.pathExists(dataDirPath) or makePath:
+            logger.debug("Image doesn't exist in archive, creating")
+            self._addImage(incremental.image, imgPath)
+            self._addMetadata(incremental.imageMetadata, metadataPath)
+
+        # Archive wasn't empty, specified paths didn't exist, and no permission
+        # given to make the path; append fails.
+        else:
+            return False
+
+        return True
+
+    @failIfEmpty
+    def getIncremental(self, sliceIndex: int = 0, **entities) \
+            -> BidsIncremental:
+        """
+        Creates a BIDS Incremental from the specified part of the archive.
+
+        Args:
+            sliceIndex: Index of 3-D image to select in a 4-D image volume.
+            entities: Keyword arguments for entities to filter by. Provide in
+                the format entity='value'.
+
+        Returns:
+            BIDS-Incremental file with the specified image of the archive and
+            its associated metadata.
+
+        Raises:
+            IndexError: If the provided sliceIndex goes beyond the bounds of the
+                image specified in the archive.
+            MissingMetadataError: If the archive lacks the required metadata to
+                make a BIDS Incremental out of an image in the archive.
+            NoMatchError: When no images that match the provided entities are
+                found in the archive
+            RuntimeError:
+                1) When too many images that match the provided entities
+                are found in the archive.
+                2) If the image matching the provided entities has fewer
+                than 3 dimensions or greater than 4.
+
+        Examples:
+            >>> archive = BidsArchive('.')
+            >>> inc = archive.getIncremental(subject='01', task='test')
+            >>> entityFilterDict = {'subject': '01', 'task': 'test'}
+            >>> inc2 = archive.getIncremental(**entityFilterDict)
+            >>> inc == inc2
+            True
+            >>> inc.imageDimensions
+            (64, 64, 27, 1)
+        """
+        if sliceIndex < 0:
+            raise IndexError(f"Slice index must be >= 0 (got {sliceIndex})")
+
+        candidates = self.getImages(**entities)
+
+        # Throw error if not exactly one match
+        if len(candidates) == 0:
+            raise NoMatchError("Unable to find any data in archive that matches"
+                               f" all provided entities: {entities}")
+        elif len(candidates) > 1:
+            raise RuntimeError("Too many results for entities (expected 1, got "
```

Comment:
GW comment: I feel this might be more of a RequestError than a runtime error

Reply:
TODO: Swap it

Reply:
TODO: Improve the error message to make clear that the entities weren't specific enough